junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/ClassBasedTestDescriptor.java

```diff
@@ -355,24 +368,36 @@ private Object invokeTestInstanceFactory(Optional<Object> outerInstance, Extensi
 	}
 
 	private Object invokeTestClassConstructor(Optional<Object> outerInstance, ExtensionRegistry registry,
-			ExtensionContext extensionContext) {
+			ExtensionContext extensionContext, ExtensionContext ourExtensionContext) {
 
 		Constructor<?> constructor = ReflectionUtils.getDeclaredConstructor(this.testClass);
 		return executableInvoker.invoke(constructor, outerInstance, extensionContext, registry,
 			InvocationInterceptor::interceptTestClassConstructor);
```

Comment:
Calls to `ParameterResolver` are not yet looking at the new annotation. Shall calls to `ParameterResolver` also consider the annotation?

I just wanted to confirm that I should adjust this method as well. I was a bit reluctant to change `InterceptingExecutableInvoker` for some reason. For now, I wrote the documentation matching the current behavior, I will change the documentation together with the implementation.

Reply:
You are right, I added `InvocationInterceptor` to the documentation and to the deliverables in the issue. I also addressed the implementation for `InvocationInterceptor`, but there are currently no tests which verify that the correct `ExtensionContext` is used. `ParameterResolver` is only covered indirectly over `TestInfo`.

Reply:
> I also addressed the implementation for `InvocationInterceptor`, but there are currently no tests which verify that the correct `ExtensionContext` is used.

Noted. I'll create an umbrella issue to sort out the remaining issues when we're done with this PR.